### ISSUE TITLE
Be able to rename Redis database

### ISSUE THREAD

**Issue by lihaif** [googleforgames/open-match#1376]
**Is your feature request related to a problem? Please describe.**

When we deploy Open Match using the helm chart, we want to rename the Redis database to avoid conflict since we already have other Redis database in the same namespace using the same name.

I can overwrite the default name using below setting in values.yaml. The database is renamed, but some Open Match service no longer work, because they still expect to use the default Redis database name.
```
open-match:
  redis:
    nameOverride: demo-redis
```

**Describe the solution you'd like**

It will be nice if Open Match supports to install Redis using a different name to the default one through configuration like the above.

**Describe alternatives you've considered**

**Additional context**

**Comment by HazWard** [comment]
The Helm chart has a dependency on the Redis helm chart from [bitnami](https://github.com/bitnami/charts/tree/master/bitnami/redis) so in your `values.yaml` you can configure the Redis settings.

From what I remember it should be something like `redis.nameOverride` but I'm not 100% sure.

**Comment by lihaif** [comment]
@HazWard Thanks for your input. As I described above, I already tried to used `redis.nameOverride`. But some Open Match services no longer work after that like below:
```
default       demo-ooc-demo-redis-master-0                        3/3     Running            0          2m7s
default       demo-ooc-demo-redis-slave-0                         3/3     Running            3          2m7s
default       demo-ooc-demo-redis-slave-1                         3/3     Running            0          76s
default       demo-ooc-open-match-backend-586d7f9b5c-pbgkk        0/1     Running            0          3h50m
default       demo-ooc-open-match-evaluator-6576bbd677-7jxvn      1/1     Running            0          3h50m
default       demo-ooc-open-match-evaluator-6576bbd677-jjw5k      1/1     Running            0          3h50m
default       demo-ooc-open-match-evaluator-6576bbd677-sr499      1/1     Running            0          3h50m
default       demo-ooc-open-match-frontend-8c4fcc6d7-fzvb8        0/1     Running            0          3h50m
default       demo-ooc-open-match-query-555bc7dcbf-54g42          0/1     Running            0          3h50m
default       demo-ooc-open-match-swaggerui-5f6f67c877-bzqq9      1/1     Running            0          3h50m
default       demo-ooc-open-match-synchronizer-76b787c667-9ptzw   0/1     Running            0          3h50m
```

**Comment by lihaif** [comment]
@HazWard, I did more investigation and find the reason. If `redis.nameOverride` is set when I install the cluster, then everything works correctly.

If I change `redis.nameOverride` after I installed the cluster and call `helm upgrade` to update the cluster, then redis is renamed, but related services(backend, frontend, query, synchronizer) do not pick up the change. If I delete the correponding pods, the new started pods pick up the `redis.nameOverride` and start to work. I suspect the reason is that related services do not pick up the config change automatically.

Thanks for your support.
